### PR TITLE
update: improve observable returning type

### PR DESCRIPTION
**Pull request by clostao**
Having the CID as optional was a weird experience. Adding distinctly typed by the field completed makes more comprehensible.

### CODE REVIEW

File: packages/auto-drive/src/api/models/uploads.ts [hunk 1]
```diff
@@ -74,17 +74,33 @@ export type CompleteUploadResponse = {
...
+      progress: number
+      cid: null
+    }
 
-export type UploadFolderStatus = {
-  type: 'folder'
-  progress: number
-  cid?: string
-}
+export type UploadFolderStatus =
```
Comment (marc-aurele-besner) [inline packages/auto-drive/src/api/models/uploads.ts:91]: Personally I find these two types a bit repetitive, why not making use of generic types?

```typescript
type UploadStatus<Type extends 'file' | 'folder'> = 
  | {
      completed: true
      type: Type
      progress: number
      cid: string
    }
  | {
      completed: false
      type: Type
      progress: number
      cid: null
    }

export type UploadFileStatus = UploadStatus<'file'>
export type UploadFolderStatus = UploadStatus<'folder'>
```

**Review by marc-aurele-besner — COMMENTED** [review]
LGTM, that been said, I find these two almost identical types a bit redundant

File: packages/auto-drive/src/api/models/uploads.ts (hunk 1 above)
Comment (clostao) [reply packages/auto-drive/src/api/models/uploads.ts:91]: Updated! Thanks

**Review by marc-aurele-besner — APPROVED** [review]
LGTM

**Comment by clostao** [comment]
Closing because is outdated

**Pull request closed without merging**